org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -413,8 +416,36 @@ private <T> void completeEmptyOrWithException(final CompletableFuture<Optional<T
 		return new LanguageServerProjectExecutor(project);
 	}
 
+	/**
+	 *
+	 * @return True if the server call made using this executor has been cancelled
```

Comment:
Does that mean that an executor can only run a single operation and then need to be discarded?